OPNsense/opnsense/ingest/parser.yml

```
@@ -0,0 +1,92 @@
+          PF_IPV4: '4,%{WORD:tos},%{DATA:ecn},%{NUMBER:hoplimit},%{NUMBER:id},%{NUMBER:offset},%{WORD:flags},%{NUMBER:protonum},%{WORD:protoname},%{NUMBER:length},%{IPV4:src},%{IPV4:dst}'
+          PF_IPV6: '6,%{DATA:class},%{DATA:flow},%{DATA:hoplimit},%{WORD:protoname},%{NUMBER:protonum},{NUMBER:length},%{IPV6:src},%{IPV6:dst}'
+          PF_UDP: '%{NUMBER:srcport},%{NUMBER:dstport},%{NUMBER:datalen}'
+          PF_TCP: '%{NUMBER:srcport},%{NUMBER:dstport},%{NUMBER:datalen},%{WORD:tcp_flags},%{NUMBER:seq},%{DATA:ack},%{NUMBER:window},%{DATA:urg},%{WORD:options}'
+          PF_CARP: '%{DATA:hoplimit},%{DATA:vhid},%{DATA:version},%{DATA:advskew},%{DATA:advbase}'
+          PF_ICMP: 'datalenght=%{DATA:icmp_datalenght}'
+
+  - name: set_ecs_fields
+    stages:
+      set_ecs_fields:
+        actions:
+          - set:
+              source.ip: "{{parsed_event.message.src}}"
+              destination.ip: "{{parsed_event.message.dst}}"
+              rule.id: "{{parsed_event.message.rulenr}}"
+              openbsd.pf.rule.subrulenr: "{{parsed_event.message.subrulenr}}"
+              rule.ruleset: "{{parsed_event.message.anchorname}}"
+              openbsd.pf.event.tracker.id: "{{parsed_event.message.label}}"
+              event.reason: "{{parsed_event.message.reason}}"
+              event.action: "{{parsed_event.message.action}}"
+              network.direction: "{{parsed_event.message.dir}}"
+              openbsd.pf.routing.class: "{{parsed_event.message.tos}}"
+              openbsd.pf.routing.hoplimit: "{{parsed_event.message.hoplimit}}"
+              openbsd.pf.routing.offset: "{{parsed_event.message.offset}}"
+              openbsd.pf.routing.flags: "{{parsed_event.message.flags}}"
+              network.network.iana_number: "{{parsed_event.message.protonum}}"
+              network.transport: "{{parsed_event.message.protoname}}"
+              network.bytes: "{{parsed_event.message.length}}"
+              source.port: "{{parsed_event.message.srcport}}"
+              destination.port: "{{parsed_event.message.dstport}}"
+              openbsd.pf.transport.bytes: "{{parsed_event.message.datalen}}"
+              openbsd.pf.transport.tcp_flags: "{{parsed_event.message.tcp_flags}}"
+              openbsd.pf.transport.sequence_number: "{{parsed_event.message.seq}}"
+              openbsd.pf.transport.ack: "{{parsed_event.message.ack}}"
+              openbsd.pf.transport.window_size: "{{parsed_event.message.window}}"
+              openbsd.pf.transport.urgency: "{{parsed_event.message.urg}}"
+              openbsd.pf.transport.options: "{{parsed_event.message.option}}"
+              event.classification: "{{parsed_event.message.class}}"
+              openbsd.pf.carp.vhid: "{{parsed_event.message.vhid}}"
+              openbsd.pf.carp.version: "{{parsed_event.message.version}}"
+              openbsd.pf.icmp.advskew: "{{parsed_event.message.advskew}}"
+              openbsd.pf.icmp.advbase: "{{parsed_event.message.advbase}}"
```

My bad: advskew and advbase are linked to CARP.
```suggestion
              openbsd.pf.carp.advskew: "{{parsed_event.message.advskew}}"
              openbsd.pf.carp.advbase: "{{parsed_event.message.advbase}}"
```